### PR TITLE
test: automation tests for tokens (search, tokens present check)

### PR DESCRIPTION
**Pull request by obmatter**
Creation of automated test cases:

https://allure.matterlabs.dev/project/17/test-cases/1436?treeId=30

https://allure.matterlabs.dev/project/17/test-cases/1564?treeId=30

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -51,19 +51,20 @@ jobs:
...
+          "@artifacts and @menuItems",
+          "@artifacts and @upperNavigationMenu",
+          "@artifacts and @contactsPage",
+          "@artifacts and @withdraw",
+          "@artifacts and @tokens",
+          "@navigation and not @emptyWallet",
+          "@transactions and @withdraw",
+          "@various and @emptyWallet",
+          "@actions and @contacts"
         ]
```
Comment (pcheremu) [inline .github/workflows/e2e.yml:68]: All these changes are necessary? Is this just an auto formatting?

File: components/common/SmallInput.vue [hunk 2]
```diff
@@ -1,20 +1,21 @@
 <template>
-  <component :is="as" class="small-input-container" :class="{ focused }">
+  <component :is="as" :class="{ focused }" class="small-input-container">
```
Comment (pcheremu) [inline components/common/SmallInput.vue:2]: You have to ask Danil for review this part of code.

File: tests/e2e/features/artifacts/depositPage/tokens.feature [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  @id1435:I @tokens
+  Scenario: Check artifacts on tokens dropdown on Deposit page (Testnet)
+    Given I go to page "/transaction/zksync/era/deposit/?network=era-goerli"
+    When I click by "testId" with "your-account" value
+    Then Element with "testId" "token-dropDown" should be "clickable"
+    Then I click by "testId" with "token-dropDown" value
+    Then Element with "text" "Choose token" should be "visible"
+    Then Element with "placeholder" "Symbol or address" should be "visible"
+
+  @id1435:I @tokens
```
Comment (pcheremu) [inline tests/e2e/features/artifacts/depositPage/tokens.feature:16]: It is make sense to use @id1435:II id as it is the 2nd part of the test

File: tests/e2e/features/artifacts/depositPage/tokens.feature [hunk 4]
```diff
@@ -0,0 +1,91 @@
+@artifacts @regression @depositPage @tokens
+Feature: Artifacts - UI
+
+  Background:
+    Given Connect Metamask extension with login action
+
+  @id1435:I @tokens
```
Comment (pcheremu) [inline tests/e2e/features/artifacts/depositPage/tokens.feature:7]: Please, add @testnet tag for testnet tests

File: tests/e2e/features/artifacts/depositPage/tokens.feature [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    Examples:
+      | TokenName | TokenAddress   | img                                                                                                                                   |
+      | ETH       | 0x000000...00A | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/eth.svg?alt=media&token=1985e3d8-3aa7-4d04-8839-565d4c341615  |
+      | DAI       | 0x3e7676...D4b | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/dai.svg?alt=media&token=1985e3d8-3aa7-4d04-8839-565d4c341615  |
+      | LINK      | 0x406091...c78 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/link.svg?alt=media&token=1985e3d8-3aa7-4d04-8839-565d4c341615 |
+      | USDC      | 0x0faF6d...2A9 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/usdc.svg?alt=media&token=1985e3d8-3aa7-4d04-8839-565d4c341615 |
+      | wBTC      | 0x0BfcE1...e9c | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/wbtc.svg?alt=media&token=1985e3d8-3aa7-4d04-8839-565d4c341615 |
+
+  @id1435:II @tokens
```
Comment (pcheremu) [inline tests/e2e/features/artifacts/depositPage/tokens.feature:33]: Please, add @mainnet tag for mainnet tests

File: tests/e2e/features/artifacts/depositPage/tokens.feature [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      | WBTC      | 0xBBeB51...011 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/wbtc.svg?alt=media&token=502e8eee-d36a-4460-a97b-dad3e387cddd                                                                                                            |
+      | KNC       | 0x6ee46C...3e6 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/wbtc.svg?alt=media&token=502e8eee-d36a-4460-a97b-dad3e387cddd                                                                                                            |
+      | BEL       | 0xB83CFB...2D9 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/bel.png?alt=media&token=f8bbcfc2-6037-4626-870b-6a3b6630554b                                                                                                             |
+      | ZZ        | 0x1ab721...184 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/zz.png?alt=media&token=dc866379-92a9-4d85-9173-dad0076046b1                                                                                                              |
+
+  @id1564 @id1436 @tokens
```
Comment (pcheremu) [inline tests/e2e/features/artifacts/depositPage/tokens.feature:80]: Also here we will need @mainnet tag (here and in general to be able to separate them next time)

**Review by pcheremu — CHANGES_REQUESTED** [review]

File: tests/e2e/features/artifacts/depositPage/tokens.feature [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  Scenario:  Check search functionality for Choose Tokens
+    Given I go to page "/transaction/zksync/era/deposit/?network=era-mainnet"
+    When I click by "testId" with "your-account" value
+    Then I click by "testId" with "token-dropDown" value
+    Then I fill the "data-testid=search_tokens" input field by "0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4"
+    Then Element with "text" "USDC" should be "visible"
+    Then I fill the "data-testid=search_tokens" input field by "0xd35CCeEAD182dcee0F148EbaC9447DAas2c4D449c4"
+    Then Element with "partial text" "Make sure you are using correct zkSync network" should be "visible"
+    Then Element with "xpath" "//*[@class='modal-card']//*[contains(., ' No tokens ')]" should be "visible"
+    Then Element with "xpath" "//*[@class='modal-card']//*[contains(., '0xd35CCeEAD182dcee0F148EbaC9447DAas2c4D449c4')]" should be "visible"
```
Comment (pcheremu) [inline tests/e2e/features/artifacts/depositPage/tokens.feature:90]: Please, make this part more human readable: 
Element with "text" "No tokens was found" should be "visible"
Element with "text" "0xd35CCeEAD182dcee0F148EbaC9447DAas2c4D449c4" should be "visible"

Comment (pcheremu) [reply tests/e2e/features/artifacts/depositPage/tokens.feature:90]: Also, please, add a new line #91

File: tests/e2e/features/artifacts/depositPage/tokens.feature [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+      | ZZ        | 0x1ab721...184 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/zz.png?alt=media&token=dc866379-92a9-4d85-9173-dad0076046b1                                                                                                              |
+
+  @id1564 @id1436 @tokens @mainnet
+  Scenario:  Check search functionality for Choose Tokens
+    Given I go to page "/transaction/zksync/era/deposit/?network=era-mainnet"
+    When I click by "testId" with "your-account" value
+    Then I click by "testId" with "token-dropDown" value
+    Then I fill the "data-testid=search_tokens" input field by "0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4"
+    Then Element with "text" "USDC" should be "visible"
+    Then I fill the "data-testid=search_tokens" input field by "0xd35CCeEAD182dcee0F148EbaC9447DAas2c4D449c4"
```
Comment (pcheremu) [inline tests/e2e/features/artifacts/depositPage/tokens.feature:87]: Can you please check if it possible to update it in this way: 
I fill the "search_tokens" input field by "0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4"
or 
I fill the element of "data-testId" with "search_tokens" value by "0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4"

File: tests/e2e/features/artifacts/depositPage/tokens.feature [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+      | KNC       | 0x6ee46C...3e6 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/wbtc.svg?alt=media&token=502e8eee-d36a-4460-a97b-dad3e387cddd                                                                                                            |
+      | BEL       | 0xB83CFB...2D9 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/bel.png?alt=media&token=f8bbcfc2-6037-4626-870b-6a3b6630554b                                                                                                             |
+      | ZZ        | 0x1ab721...184 | https://firebasestorage.googleapis.com/v0/b/token-library.appspot.com/o/zz.png?alt=media&token=dc866379-92a9-4d85-9173-dad0076046b1                                                                                                              |
+
+  @id1564 @id1436 @tokens @mainnet
+  Scenario:  Check search functionality for Choose Tokens
```
Comment (pcheremu) [inline tests/e2e/features/artifacts/depositPage/tokens.feature:81]: Can you please split positive and negative scenarios.  Like @id1564:I and @id1564:II 
for positive you will be able to check ERC20 token as well as any other custom. and for negative you will be able to check any other tricky scenarios.

**Review by pcheremu — CHANGES_REQUESTED** [review]

**Comment by obmatter** [comment]
@pcheremu fixed all the comments - please take a look.
Thank you in advance!

**Pull request closed without merging**